### ISSUE TITLE
Vaccine hierarchy, low hanging fruits

### ISSUE THREAD

**Issue by dimshitc** [OHDSI/Vocabulary-v5.0#763]
I'm looking at mapping changes, for example 
this [CPT Human Papillomavirus vaccine ... ](https://athena.ohdsi.org/search-terms/terms/2213435) was mapped to [CVX](https://athena.ohdsi.org/search-terms/terms/40213320), now it's mapped to [RxNorm](https://athena.ohdsi.org/search-terms/terms/529079), which is a correct mapping.
There was a big problem with this systematic approach with tons of attributes when connecting RxNorm to CVX.
But I would do it manually at least for those that can be done. There are around 200 of CVX codes, so it's doable.
You can start from the HCPCS or CPT mappings changed - you already have CVX and RxNorm. (you just go up from that RxNorm and find closest concepts to CVX)
In this case this CVX becomes a parent of this RxNorm https://athena.ohdsi.org/search-terms/terms/36230392 and https://athena.ohdsi.org/search-terms/terms/36223103

**Comment by Alexdavv** [comment]
@dimshitc As we discussed [here](https://forums.ohdsi.org/t/preparing-for-covid-vaccine-codes-in-the-vocabulary-and-modeling-in-the-omop-cdm/12446/12), it only works unless they introduced a new vaccine with a different set of strains, dose or whatever else. Which will be the same or, more likely, a wholly new CVX that will break the hierarchy you suggested.

It's happening all over the place with COVID and we really don't have a good solution for old coding like [this](https://forums.ohdsi.org/t/best-mapping-for-specific-covid-vaccines/18047). The same could happen in each place where they actively develop. So manual mapping and further maintenance don't look like a reliable solution.

Also, you know better than me that the team has been working on this for a while trying to push it out of the doors [here](https://forums.ohdsi.org/t/vaccines-standard-cvx-or-rxnorm/6224/12) and [here](https://forums.ohdsi.org/t/proposal-of-cvx-and-cvx-rxnorm-cvx-atc-hierarchical-crosswalks-creation/6921). And then other folks from the Vaccine vocabulary WG also [did](https://ohdsiorg.sharepoint.com/:w:/s/Workgroup-VaccineVocabulary/EZLHOMLilX5Eqb9890uiPkkB4IH2e9B61JXFqQuWdxVIvw?e=Fh6Xx7).

As you explicitly mentioned, the tons of attributes in the way they're randomly spread over the Standard vocabularies, will not allow us to use the current system. Even though we'll add a few manually curated links, can you rely on this hierarchy in your research? No, you still need to go through the whole bunch of things in multiple vocabularies, including the sources (because the mappings are nasty).

The only performing solution would be a new and nice vaccine vocabulary. The vaccine vocabulary WG has just started over trying to use the [Vaccine ontology](https://violinet.org/vaccineontology/) as the tool that would help to put things together. Please join each other Monday 9 AM EST.

@yongqunh

**Comment by yongqunh** [comment]
Yes. This is the type of work that our new vaccine vocabulary WG plans to do this year. Welcome to join our biweekly meetings!

Oliver He

**Comment by dimshitc** [comment]
In the example above there's exact strains mentioned anyway, so this case stays within same RxNorm and CVX codes even if they invent new Human Papillomavirus vaccine. 
And regarding generic CVX codes, do we really believe they will invent new vaccines? That was always our obstacle, that hypothetically the new vaccine will be invented and it be thrown out the hierarchy. But I would look at the real world rather than theoretical possibilities.
Anyway, I would be happy to learn more about the vaccine ontology and how are you going to use it. Do you have any presentation or overview? Or maybe you plan to present to on the one of the WG calls?
And thank you for the invitation to the WG:)
